Sprint_2

Djinslayer opened the pull request:
1. В pom.xml подключены Jacoco, Mockito и JUnit.
2. Класс Lion изолирован от класса Feline.
3. Созданы тесты на классы Feline, Cat и Lion.
4. Сделан отчёт с помощью Jacoco

Commit:
1. В pom.xml подключены Jacoco, Mockito и JUnit.
2. Класс Lion изолирован от класса Feline.
3. Созданы тесты на классы Feline, Cat и Lion.
4. Сделан отчёт с помощью Jacoco

alesyaisachenko commented on src/main/java/com/example/Lion.java:
```diff
@@ -1,32 +1,36 @@
+
+    private Feline feline;
+
+    public Lion(String sex) throws Exception {
```
изоляцию Lion от Feline нужно было делать через инъекцию зависимостей. То есть в класс Lion должен попадать уже созданный экземпляр класса Feline.

В конструкторе необходимо задавать feline

Это нужно исправить.

alesyaisachenko commented on src/test/java/com/example/FelineTest.java:
```diff
@@ -0,0 +1,86 @@
+        assertEquals("Ожидается список еды для травоядных", expectedFood, actualFood);
+    }
+
+    @Test
```
Здесь лучше было реализовать так:
@Test(expected = Exception.class)

alesyaisachenko commented on src/test/java/com/example/FelineTest.java:
```diff
@@ -0,0 +1,86 @@
+    @Test
+    public void getFoodPredatorReturnsCorrectList() throws Exception {
+        //Arrange
+        Feline feline = new Feline();
```
Feline feline = new Feline(); -- этот код можно было вынести в @Before.
Таким образом ты бы избежал дублирование кода.

Можно не править.

alesyaisachenko commented on src/test/java/com/example/FelineTest.java:
```diff
@@ -0,0 +1,86 @@
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
```
в этом классе ты не используешь моки

Commit:
1. Изоляция Lion от Feline сделана через инъекцию зависимостей.
2. Из FelineTest убраны упоминания о моках.
3. Сделан новый отчёт с помощью Jacoco.
